### ISSUE TITLE
[libprotobuf ERROR google/protobuf/io/coded_stream.cc:207] A protocol message was rejected because it was too big

### ISSUE THREAD

**Issue by Chipcui** [vgteam/vg#2954]
**1. What were you trying to do?**
First, I used seqwish to build a GFA format graph:
```
minimap2 westar_kale.chrA01.fa.gz westar_kale.chrA01.fa.gz -c -X -x asm5 -t $THREADS -o westar_kale.chrA01.paf
seqwish -s westar_kale.chrA01.fa.gz -t $THREADS -b westar_kale.chrA01.graph -p westar_kale.paf -g westar_kale.chrA01.gfa
```
Then, I convert the GFA to vg:
```
vg convert -g westar_kale.chrA01.gfa -x -t 1 >westar_kale.chrA01.xg
```
And I make an index :
```
vg index -L -x westar_kale.chrA01.v2.xg westar_kale.chrA01.vg
```

**2. What did you want to happen?**

**3. What actually happened?**
When I make an index of the vg, it occurs the error:
```
[libprotobuf ERROR google/protobuf/io/coded_stream.cc:207] A protocol message was rejected because it was too big (more than 67108864 bytes).  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
error[load_proto_to_graph]: invalid Graph message
```

**4. If you got a line like `Stack trace path: /somewhere/on/your/computer/stacktrace.txt`, please copy-paste the contents of that file here:**

```
Place stacktrace here.
```

**5. What data and command can the vg dev team use to make the problem happen?**

**6. What does running `vg version` say?**

```
vg version v1.25.0 "Apice"
Compiled with g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0 on Linux
Linked against libstd++ 20191114
Built by anovak@octagon
```

Thank you in advance!

**Comment by jmonlong** [comment]
I'm having similar issues. It seems to be something with the paths (too long?).

As suggested in #2772 I found that it helps to use the HashGraph format. You could try the `-a` option for the conversion:

```
vg convert -g westar_kale.chrA01.gfa -a >westar_kale.chrA01.vg
vg index -L -x westar_kale.chrA01.v2.xg westar_kale.chrA01.vg
```

I'm still having problems with some vg commands. For example `vg augment` won't augment long paths, even if the input graph is a HashGraph. Is augment maybe internally going back to protobuf @glennhickey ?

```
vg augment hashgraph.vg paths.gam
# [libprotobuf ERROR google/protobuf/io/coded_stream.cc:207] A protocol message was rejected because it was too big (more than 67108864 bytes).  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
vg augment -B hashgraph.vg paths.gam
# works
```

**Comment by glennhickey** [comment]
That is strange.  The only protobuf in augment should be the GAM, which is
the same whether or not you use -B.  Also, the message size limit in vg has
been 2G for a while to get around these errors.  Did it somehow get
reverted back down to 64G?   Could you please share the data?

**Comment by jmonlong** [comment]
I was trying to do this using the latest release (v1.26.1):

```
vg paths -X -x hpp60-hg38.minigraph.L50-l50k.xg -g hpp60-hg38.minigraph.L50-l50k.N16.gbwt > paths.gam
vg augment -t 4 -i hpp60-hg38.minigraph.L50-l50k.vg paths.gam > aug.vg
```

Using

s3://vg-k8s/vgamb/chr20/minigraph/L50-l50k/hpp60-hg38.minigraph.L50-l50k.vg
s3://vg-k8s/vgamb/chr20/minigraph/L50-l50k/map/hpp60-hg38.minigraph.L50-l50k.xg
s3://vg-k8s/vgamb/chr20/minigraph/L50-l50k/map/hpp60-hg38.minigraph.L50-l50k.N16.gbwt

Although this .vg might not be HashGraph, I tried to convert it to HashGraph and got the same error.

```
[libprotobuf ERROR google/protobuf/io/coded_stream.cc:207] A protocol message was rejected because it was too big (more than 67108864 bytes).  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
terminate called after throwing an instance of 'std::runtime_error'
  what():  obsolete, invalid, or corrupt protobuf input
```

**Comment by glennhickey** [comment]
thanks.  taking a look.  fwiw you can verify the format of a graph with "vg
stats -F"

**Comment by glennhickey** [comment]
Yeah, it seems like the old limit that was [set in vg's stream.hpp](#1718) was moved over during a refactor [into message_iterator.hpp](https://github.com/vgteam/libvgio/blob/d6e1811841f74ba04735cf92e5613a26008c83d5/include/vg/io/message_iterator.hpp#L40-L43) and is used in [protobuf_iterator.hpp](https://github.com/vgteam/libvgio/blob/d6e1811841f74ba04735cf92e5613a26008c83d5/include/vg/io/protobuf_iterator.hpp#L237-L241).  This is all well and good, but the code reading the GAM, at least in this case, is `for_each_parallel_impl` in the new stream.hpp, and it's reading [messages directly using ParseFromString()](https://github.com/vgteam/libvgio/blob/d6e1811841f74ba04735cf92e5613a26008c83d5/include/vg/io/stream.hpp#L205-L211) which bypasses the CodedMessageStream interface that applies the limit. 

tldr: I think this is a regression that happened somewhere in the libvgio refactor and, especially given that we still depend on GAM, will need to somehow be fixed.  @adamnovak what do you think?

**Comment by glennhickey** [comment]
Perhaps GAF could be a work-around here?  Would probably need to be added to paths and augment, but may be faster than fixing the gam reader...

**Comment by glennhickey** [comment]
@jmonlong It seems the protobuf message size issue only happens when streaming a GAM in parallel.  This is why `augment` works with `-B` and not without.  With #2976 you can disable parallel streaming with `vg augment -t 1`.  You can also work around it with the vg version you have by using `vg augment -N 1`.  This will turn off the N filter, but if that's not a problem, then you should be okay.

**Comment by jmonlong** [comment]
Thanks, that's going to be more elegant than the workaround I was using (going through GFA, adding the paths and then back to .vg).

**Comment by adamnovak** [comment]
I think we need to tweak the code that uses `ParseFromString` to up the limit as well. When setting the limit we were worried about big graphs, not big GAM records. 64 MB is a big alignment.

**Comment by adamnovak** [comment]
OK, I replaced all the `ParseFromString` calls outside the old RocksDB indexing system with calls to a function that should set the limit higher, in #2982. I haven't tested with the example files from S3 yet though.

**Comment by glennhickey** [comment]
@jmonlong Also #2983 added GAF support to `vg paths` and `vg augment` so you should be able to bypass GAM altogether now if you want.
